src/urllib3/util/ssltransport.py

```diff
@@ -89,15 +124,20 @@ def sendall(self, data, flags=0):
                 v = self.send(byte_view[count:])
                 count += v
 
-    def send(self, data, flags=0):
+    def send(self, data: bytes, flags: int = 0) -> int:
         if flags != 0:
             raise ValueError("non-zero flags not allowed in calls to send")
-        response = self._ssl_io_loop(self.sslobj.write, data)
-        return response
+        return self._ssl_io_loop(self.sslobj.write, data)
 
     def makefile(
-        self, mode="r", buffering=None, encoding=None, errors=None, newline=None
-    ):
+        self,
+        mode: Union["_TextMode", "_BinaryMode"],
```

Comment:
Since the `socket` object is transparent to users do we need this level of granularity for our type hints (ie is it okay to use `str` here?)

Reply:
`makefile` implementation already checks if mode contains only letters 'b', 'r' and 'w'. I think one thing that could go unnoticed is having duplicates like 'rrw' but it is rather unlikely so for simplicity I think it is ok to use `str`.